docs/Customization/Firrtl-Transforms.rst

```diff
@@ -0,0 +1,73 @@
+.. _firrtl-transforms:
+
+Adding a Firrtl Transform
+=========================
+
+After writing the Chisel RTL, you can make further modifications by adding transforms during the FIRRTL compilation phase.
```

Comment:
Should make analogy to LLVM transforms for software.
"Similar to how LLVM IR passes can perform transformations and optimizations on software, FIRRTL transforms can modify Chisel-elaborated RTL."